File: ANA SOFIA BARREA GIRALDO/EJERCICIOS2.CS/ejercicio_p03.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        Dictionary<string, double> productos = new Dictionary<string, double>();

        productos["Arroz"] = 4000;
        productos["Leche"] = 3500;
        productos["Pan"] = 2000;

        while (true)
        {
            Console.WriteLine("\n--- MENU ---");
            Console.WriteLine("1. Agregar producto");
            Console.WriteLine("2. Consultar precio");
            Console.WriteLine("3. Actualizar precio");
            Console.WriteLine("4. Eliminar producto");
            Console.WriteLine("5. Salir");

            Console.Write("Seleccione una opción: ");
            string opcion = Console.ReadLine();

            if (opcion == "1")
            {
                Console.Write("Ingrese el producto: ");
                string producto = Console.ReadLine();

                Console.Write("Ingrese el precio: ");
                double precio = double.Parse(Console.ReadLine());

                productos[producto] = precio;

                Console.WriteLine("Producto agregado correctamente.");
            }
            else if (opcion == "2")
            {
                Console.Write("Ingrese el producto: ");
                string producto = Console.ReadLine();

                if (productos.ContainsKey(producto))
                {
                    Console.WriteLine("El precio es: " + productos[producto]);
                }
                else
                {
                    Console.WriteLine("Producto no encontrado.");
                }
            }
            else if (opcion == "3")
            {
                Console.Write("Ingrese el producto: ");
                string producto = Console.ReadLine();

                if (productos.ContainsKey(producto))
                {
                    Console.Write("Ingrese el nuevo precio: ");
                    double precio = double.Parse(Console.ReadLine());

                    productos[producto] = precio;

                    Console.WriteLine("Precio actualizado.");
                }
                else
                {
                    Console.WriteLine("Producto no encontrado.");
                }
            }
            else if (opcion == "4")
            {
                Console.Write("Ingrese el producto: ");
                string producto = Console.ReadLine();

                if (productos.ContainsKey(producto))
                {
                    productos.Remove(producto);
                    Console.WriteLine("Producto eliminado.");
                }
                else
                {
                    Console.WriteLine("Producto no encontrado.");
                }
            }
            else if (opcion == "5")
            {
                Console.WriteLine("Programa terminado.");
                break;
            }
            else
            {
                Console.WriteLine("Opción no válida.");
            }
        }
    }
}

File: ANA SOFIA BARREA GIRALDO/EJERCICIOS.PY/ejercicio_p03.py
productos = {
    "Arroz": 4000,
    "Leche": 3500,
    "Pan": 2000
}

while True:
    print("\n--- MENU ---")
    print("1. Agregar producto")
    print("2. Consultar precio")
    print("3. Actualizar precio")
    print("4. Eliminar producto")
    print("5. Salir")

    opcion = input("Seleccione una opción: ")

    if opcion == "1":
        producto = input("Ingrese el producto: ")
        precio = float(input("Ingrese el precio: "))

        productos[producto] = precio
        print("Producto agregado correctamente.")

    elif opcion == "2":
        producto = input("Ingrese el producto: ")

        if producto in productos:
            print("El precio es:", productos[producto])
        else:
            print("Producto no encontrado.")

    elif opcion == "3":
        producto = input("Ingrese el producto: ")

        if producto in productos:
            precio = float(input("Ingrese el nuevo precio: "))
            productos[producto] = precio
            print("Precio actualizado.")
        else:
            print("Producto no encontrado.")

    elif opcion == "4":
        producto = input("Ingrese el producto: ")

        if producto in productos:
            del productos[producto]
            print("Producto eliminado.")
        else:
            print("Producto no encontrado.")

    elif opcion == "5":
        print("Programa terminado.")
        break

    else:
        print("Opción no válida.")